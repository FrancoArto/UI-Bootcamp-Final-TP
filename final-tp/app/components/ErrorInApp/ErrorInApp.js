'use strict';
import React, { Component } from 'react';
import { Text, View } from 'react-native';
import styles from './errorInApp.style';


class ErrorInApp extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
          <View style={styles.container}>
            <View style={styles.errorContainer}>
                <Text style={styles.errorMessage}>Ups! we have a error in : {this.props.errorMessage.toString()} </Text>
            </View>
          </View>
        );
    }
};

export default ErrorInApp;