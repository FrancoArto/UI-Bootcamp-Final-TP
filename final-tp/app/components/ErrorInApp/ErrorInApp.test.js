import React from 'react';

import ShallowRenderer from 'react-test-renderer/shallow';

import ErrorInApp from './ErrorInApp';


const error = 'ERROR MESSAGE'
it('renders without crashing', () => {
  const rendered = new ShallowRenderer()
  rendered.render(<ErrorInApp errorMessage={error} />)
  expect(rendered).toMatchSnapshot();
});