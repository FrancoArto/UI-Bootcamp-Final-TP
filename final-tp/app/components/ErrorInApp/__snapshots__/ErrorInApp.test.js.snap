// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ReactShallowRenderer {
  "_context": Object {},
  "_element": <ErrorInApp
    errorMessage="ERROR MESSAGE"
  />,
  "_forcedUpdate": false,
  "_instance": ErrorInApp {
    "context": Object {},
    "props": Object {
      "errorMessage": "ERROR MESSAGE",
    },
    "refs": Object {},
    "state": null,
    "updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
  },
  "_newState": null,
  "_rendered": <Component
    style={
      Object {
        "backgroundColor": "#cc3132",
        "flex": 1,
      }
    }
  >
    <Component
      style={
        Object {
          "marginLeft": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        Ups! we have a error in : 
        ERROR MESSAGE
         
      </Text>
    </Component>
  </Component>,
  "_rendering": false,
  "_updater": Updater {
    "_callbacks": Array [],
    "_renderer": [Circular],
  },
}
`;
